api/src/pcapi/routes/backoffice/offers/blueprint.py

```diff
@@ -399,14 +399,22 @@ def _get_offers(form: forms.InternalSearchForm) -> list[offers_models.Offer]:
 def _get_advanced_search_args(form: forms.InternalSearchForm) -> dict[str, typing.Any]:
     advanced_query = ""
     search_data_tags = set()
+
     if form.search.data:
         advanced_query = f"?{request.query_string.decode()}"
 
         for data in form.search.data:
-            if data.get("operator"):
+            if search_operator := data.get("operator"):
                 if search_field := data.get("search_field"):
                     if search_field_attr := getattr(forms.SearchAttributes, search_field, None):
-                        search_data_tags.add(search_field_attr.value)
+                        field_name = SEARCH_FIELD_TO_PYTHON[search_field]["field"]
+                        field_data = data[field_name] if isinstance(data[field_name], list) else [data[field_name]]
```

Comment:
Est-ce qu'on est sûr que si on a une collection, c'est forcément une liste et pas un tuple, un set ou un générateur ?

Reply:
`data` est forcément de type `list`, ça vient d'un champ de type `PCSelectMultipleField` qui hérite de [SelectMultipleField](https://wtforms.readthedocs.io/en/2.3.x/fields/#wtforms.fields.SelectMultipleField) et dans `process_data` où `self.data` est attribué on a explicitement une liste : [self.data = list(self.coerce(x) for x in valuelist)](https://github.com/wtforms/wtforms/blob/master/src/wtforms/fields/choices.py#L184)
Idem pour `PCTomSelectField`

Reply:
Si `data[field_name]` est forcément une liste, pourquoi ce `if`/`else` ? Désolé je ne comprends pas bien =/

Reply:
`data[field_name]` est forcément une liste pas un tuple, set, generator ou autre.
Le but de cette ligne c'est d'uniformiser la façon dont vont être traités les valeurs des champs du formulaire pour les afficher dans la page des offres.
Donc si la valeur du champ est une liste on veut afficher l'ensemble de ses éléments séparés par `", "` sinon, on affiche sa représentation. 
Le but de cette ligne est d'obtenir une liste de valeurs peu importe le type des données pour pouvoir l'utiliser plus bas dans la méthode `str.join`.